tests/helpers.py

```diff
@@ -59,11 +59,19 @@ def upload_deb_package(
 
 
 def check_installed(conn, pkg, installed=True):
-    """Check whether the given package is installed on the device given by
-    conn."""
+    """Check whether the given package is installed on the device given by conn.
+    Check the specific dpkg Status to differentiate between installed (install ok installed)
+    and other status like removed but not purged (deinstall ok config-files)"""
 
     res = conn.run(f"dpkg --status {pkg}", warn=True)
     if isinstance(res, FabricResult):
-        assert (res.return_code == 0) == installed
+        retcode = res.return_code
+        output = res.stdout
```

Comment:
Is this Python 2..? :open_mouth:

Reply:
I don't understand. What exactly is Python 2, the `isinstance`?

Reply:
I'm talking about the missing `decode`. But is it just a difference in Fabric maybe? I guess I'm not really clear on when this is a `FabricResult` and when it isn't.

Reply:
Right. Yes, so `FabricResult` is using strings. The other case is a result from `subprocess.check_output`. Hopefully at some point we will remove fabric also from this test framework :)